### ISSUE TITLE
Can not reach gateway when set network gateway

### ISSUE THREAD

**Issue by qkboy** [contiv/netplugin#335]
I found the container can not reach external gateway when create the contiv network with gateway. But if the network without gateway, the container can reach it.
example:
I build two vlan in the switch with two gateway.
```
vlan id:3150 <--> gw: 172.16.150.1
vlan id:3151 <--> gw: 172.16.151.1
```
Create two network:
```
[root@docker-contiv01 ~]# /apps/svr/netplugin/netctl net ls
Tenant   Network   Encap type  Packet tag  Subnet           Gateway
------   -------   ----------  ----------  -------          ------
default  vlan3150  vlan        3150        172.16.150.0/24  
default  vlan3151  vlan        3151        172.16.151.0/24  172.16.151.1
```

Create two container connect to each above network:
```
[root@docker-contiv01 ~]# docker -H :10000  run -d --name docker-node01 --privileged --net vlan3150 sshdserver:v1  
[root@docker-contiv02 ~]# docker inspect docker-node01 |grep IPAddress
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.16.150.3",

[root@docker-contiv01 ~]# docker -H :10000  run -d --name docker-vlan151-01 -h docker-vlan151-01 --privileged --net vlan3151 sshdserver:v1 
[root@docker-contiv03 ~]# docker inspect docker-vlan151-01 |grep IPAddress
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.16.151.3",

skydns container:
[root@docker-contiv01 ~]# docker inspect defaultdns|grep IPAddress 
            "SecondaryIPAddresses": null,
            "IPAddress": "172.17.0.4",
                    "IPAddress": "172.17.0.4",
                    "IPAddress": "172.16.150.2",
                    "IPAddress": "172.16.151.2",
```

Both container can reach defaultdns skydns container. That means physical switch already set trunk and allow vlan id 3150/3151 .
```
[root@docker-contiv02 ~]# docker exec docker-node01 ping -c2 172.16.150.2
PING 172.16.150.2 (172.16.150.2) 56(84) bytes of data.
64 bytes from 172.16.150.2: icmp_seq=1 ttl=64 time=4.33 ms
64 bytes from 172.16.150.2: icmp_seq=2 ttl=64 time=0.363 ms

[root@docker-contiv03 ~]# docker exec docker-vlan151-01 ping -c2 172.16.151.2
PING 172.16.151.2 (172.16.151.2) 56(84) bytes of data.
64 bytes from 172.16.151.2: icmp_seq=1 ttl=64 time=3.90 ms
64 bytes from 172.16.151.2: icmp_seq=2 ttl=64 time=0.454 ms
```

Only vlan3150 network can reach the external gw:
(Turn off eth1 that connect to docker docker_gwbridge interface. )
```
[root@docker-contiv02 ~]# docker exec docker-node01 ping -c2 172.16.150.1
PING 172.16.150.1 (172.16.150.1) 56(84) bytes of data.
64 bytes from 172.16.150.1: icmp_seq=1 ttl=254 time=18.6 ms
64 bytes from 172.16.150.1: icmp_seq=2 ttl=254 time=0.730 ms

[root@docker-contiv02 ~]# docker exec -it docker-node01 /bin/bash 
[root@5c15f9703125 /]# arp -n
Address                  HWtype  HWaddress           Flags Mask            Iface
172.16.150.1             ether   9c:37:f4:7a:df:d2   C                     eth0 <-- external gw
172.16.150.2             ether   02:02:ac:10:96:02   C                     eth0
```

container on vlan3150 can't.
```
[root@docker-contiv03 ~]# docker exec docker-vlan151-01 ping -c2 172.16.151.1
PING 172.16.151.1 (172.16.151.1) 56(84) bytes of data.
From 172.16.151.3 icmp_seq=1 Destination Host Unreachable
From 172.16.151.3 icmp_seq=2 Destination Host Unreachable

--- 172.16.151.1 ping statistics ---
2 packets transmitted, 0 received, +2 errors, 100% packet loss, time 3006ms
pipe 2

```
Why? Is it anything not set or create? THX.

**Comment by shaleman** [comment]
@qkboy We fixed a gateway ARP issue recently. Please see #321 
This issue sounds exactly same as the one fixed in #321. 
Do you know what version on netplugin are you running on your server?
you can do "netctl version" to get the version info

**Comment by qkboy** [comment]
```
[root@docker-contiv01 ~]# /apps/svr/netplugin/netctl version
Client Version:
Version: v0.1-04-09-2016.02-54-52.UTC
GitCommit: 54d3775
BuildTime: 04-09-2016.02-54-52.UTC

Server Version:
Version: v0.1-04-09-2016.02-54-52.UTC
GitCommit: 54d3775
BuildTime: 04-09-2016.02-54-52.UTC
```
We fixed them with if dstEsp is gateway ,then reinject to uplink port and send arp request to external switch for search gateway mac.
```
contiv/ofnet/vlanBridge.go：
fmt.Println("The src and des ep are", srcEp, dstEp)
if dstEp != nil && ( dstEp.MacAddrStr == "") {
                                dstEp = nil
}
```
Test is ok.

**Comment by shaleman** [comment]
@qkboy 
this change would work too. But, #321 fixes it generically.
I'm going to close this bug since its fixed already by #321.

**Comment by shaleman** [comment]
fixed by #321
